torchtune/datasets/_chat.py

```diff
@@ -167,15 +120,15 @@ def chat_dataset(
         ValueError: if the conversation format is not supported
     """
     if conversation_format == "sharegpt":
-        convert_to_dialogue = sharegpt_to_llama2_dialogue
+        convert_to_messages = sharegpt_to_llama2_messages
     else:
         raise ValueError(f"Unsupported conversation format: {conversation_format}")
 
     return ChatDataset(
         tokenizer=tokenizer,
         source=source,
-        convert_to_dialogue=convert_to_dialogue,
-        template=_get_template(template),
+        convert_to_messages=convert_to_messages,
+        chat_format=chat_format,
```

Comment:
This needs a mapping from str to the actual class pointer, a simpler version of get_template

Reply:
I am just gonna pass the `ChatFormat` directly for now, lmk if that makes sense

Reply:
hm, then this builder will require a nested component to use from the config. It won't work from the config then. I can update it in a follow up if needed

Reply:
will leave here and duck - is the one layer of component instantiation coming in the way of getting stuff done?

Reply:
Yeah let's do in a follow-up. This will only affect custom chat datasets, right? Which we don't have any of yet anyways. Re nested instantiation, imo this is not a sufficient reason to add it.. I think we can find another way here

Reply:
I think the string mapping is very simple and uses our existing tools without requiring nested instantiation or impacting UX. Will do a follow-up and we can discuss there.